### PR TITLE
feat: Add support for generating workspace API keys

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for generating workspace API keys

## Motivation and Context
- Provide support for generating API keys following release of https://github.com/hashicorp/terraform-provider-aws/pull/25286

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -29,6 +29,19 @@ resource "aws_grafana_workspace" "this" {
   tags = var.tags
 }
 
+################################################################################
+# Workspace API Key
+################################################################################
+
+resource "aws_grafana_workspace_api_key" "this" {
+  for_each = { for k, v in var.workspace_api_keys : k => v if var.create }
```
Comment (antonbabenko) [inline main.tf:37]: Maybe it should check var.create_workspace instead?

Comment (bryantbiggs) [reply main.tf:37]: [this](https://github.com/terraform-aws-modules/terraform-aws-managed-service-grafana/pull/9/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR42) handles that but slightly differently. 

tl;dr - users can create a workspace and all the things here, or they can create all the things here (minus a workspace) on an existing workspace

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 1.5.0 :tada:
